common/src/main/java/com/microsoft/identity/common/internal/platform/DevicePopManager.java

```diff
@@ -615,6 +626,163 @@ public boolean verify(@NonNull final SigningAlgorithm alg,
         return false;
     }
 
+    @Override
+    public String encrypt(@NonNull final Cipher cipher,
+                          @NonNull final String plaintext) throws ClientException {
+        final String methodName = ":encrypt";
+        final String errCode;
+        final Exception exception;
+
+        try {
+            // Load our key material
+            final KeyStore.PrivateKeyEntry privateKeyEntry = (KeyStore.PrivateKeyEntry)
+                    mKeyStore.getEntry(mKeyAlias, null);
+
+            // Get a ref to our public key
+            final PublicKey publicKey = privateKeyEntry.getCertificate().getPublicKey();
+
+            // Init our Cipher
+            final javax.crypto.Cipher input = javax.crypto.Cipher.getInstance(cipher.toString());
+            input.init(javax.crypto.Cipher.ENCRYPT_MODE, publicKey);
+
+            // Declare an OutputStream to hold our encrypted data
+            final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+            // Create a B64Stream to encode our incoming data, and write it to our ByteArrayStream
+            final Base64OutputStream base64OutputStream = new Base64OutputStream(
+                    byteArrayOutputStream,
+                    Base64.DEFAULT
+            );
+
+            // Wrap it in our CipherOutputStream, write the contents...
+            final OutputStream cipherOutputStream = new CipherOutputStream(base64OutputStream, input);
+            cipherOutputStream.write(plaintext.getBytes(ENCODING_UTF8));
+            cipherOutputStream.close();
```

Comment:
Close in a finally block to avoid any potential resource leaks.

Reply:
ACK

Reply:
da7aff83fe5996f26db6e0af75a48bbf2e2183cb